### DISCUSSION TITLE
NTFS mount issues on rAudio-RPi0-1-20240615 with Raspberry Pi Zero W

### DISCUSSION THREAD

**Discussion by MulberryPi** [rern/rAudio#1734] [Bug Reports]
Hello all,

rAudio-RPi0-1-20240615 with Raspberry Pi Zero W seems to refuse to mount my NTFS USB flash drive(s), however if I take the same USB to a  i20240615/rAudio-64bit-20240615 on Raspberry Pi 3, the drive(s) seem to mount perfectly fine

I am creating the NTFS drive in Windows 10 before copying the music to it, and run chkdsk which shows no issues. 

Below is an excerpt from the startup which shows the "denied 73" error (tried multiple USB flash drives and always get same error)

Jun 17 21:29:07 rAudioZero devmon[216]: added:     /org/freedesktop/UDisks/devices/sda1
Jun 17 21:29:08 rAudioZero devmon[216]: device: [/dev/sda1]
Jun 17 21:29:08 rAudioZero devmon[216]:     systeminternal: [0]
Jun 17 21:29:08 rAudioZero devmon[216]:     usage:          [filesystem]
Jun 17 21:29:08 rAudioZero devmon[216]:     type:           [ntfs3]
Jun 17 21:29:08 rAudioZero avahi-daemon[219]: Got SIGHUP, reloading.
Jun 17 21:29:08 rAudioZero devmon[216]:     label:          [2GB]
Jun 17 21:29:08 rAudioZero devmon[216]:     ismounted:      [0]
Jun 17 21:29:08 rAudioZero devmon[216]:     nopolicy:       [0]
Jun 17 21:29:08 rAudioZero devmon[216]:     hasmedia:       [1]
Jun 17 21:29:08 rAudioZero devmon[216]:     opticaldisc:    []
Jun 17 21:29:08 rAudioZero devmon[216]:     numaudiotracks: []
Jun 17 21:29:08 rAudioZero devmon[216]:     blank:          []
Jun 17 21:29:08 rAudioZero devmon[216]:     media:          []
Jun 17 21:29:08 rAudioZero devmon[216]:     partition:      [1]
Jun 17 21:29:08 rAudioZero devmon[216]: devmon: /usr/bin/udevil --mount /dev/sda1 --mount-options "noexec,nosuid,nodev,noatime"    # (2GB)
Jun 17 21:29:08 rAudioZero dhcpcd[286]: wlan0: adding default route via 192.168.1.254
Jun 17 21:29:08 rAudioZero devmon[216]: udevil: denied 73: fstype 'ntfs3' is not an allowed type
Jun 17 21:29:08 rAudioZero devmon[216]: devmon: error mounting /dev/sda1 (2)

Any ideas on this one?

**Comment by ceejayemm** [comment]
rAudio prefers Linux file systems (EXT4 etc) and supports file sharing via SAMBA (Features / Others / File Sharing) - so could you format the USB to say EXT4  and mount it on the RPi, sharing it via Samba to your Windows machines to allow copying og the files you want on the USB drive via the Samba share ?  

Or go the other way, create a share on your Windows machine and share it to the RPi.  rAudio can then mount a CIFS or NFS share (or more if needed) to access the files on the share(s) ?  This is what I do, my music is held on a NAS and shared via CIFS/NFS to the RPI. In this way I can connect several Audio installations to one single source of music.

In both cases Samba / CIFS / NFS will deal with the 'translations' between the differing native file systems.

Chris

**Reply by MulberryPi** [reply]
Thanks Chris, appreciate the suggestions and will move in that direction if I can't get past this one. 

It would be very nice to use local USB library for this application though so I'm curious if it's a simple config file change or command to run on the system where it "fixes" the file system etc...

**Reply by ceejayemm** [reply]
What happens if you format the USB drive to something like exFAT, will it mount then ?  If it will then you should be able to copy files from your Windows machine to that for use in rAudio.

Chris

**Reply by fscara** [reply]
Chances are that the problem is not in rAudio but in the Linux drivers. 
There are many good reasons NTFS support is not as good as exFAT/CIFS support.
f.

**Comment by MulberryPi** [comment]
Thanks all! exFAT is working and issue resolved!
